### PR TITLE
docs: Document select types and select channel types

### PR DESCRIPTION
**Pull request by aLEGEND21**
## Summary

Documented select types and select channel types in the guide.

## Information

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by Lulalaby** [comment]
Thank you in advance! We're gonna look at some point the next days over this pr

File: docs/interactions/ui-components/dropdowns.mdx [hunk 1]
```diff
@@ -100,6 +102,39 @@ In the `options` parameter, you pass a list of [`discord.SelectOption`](https://
...
 
+## Select Types
+
+In addition to regular string selects, you can also have your select menu contain users, roles, mentionables, and channels as its options. You can use these alternative select types by passing a [`discord.ComponentType`](https://docs.pycord.dev/en/stable/api/enums.html#discord.ComponentType) value for the `select_type` parameter when creating the Select.
+
+:::info Viewing the selected option
+
+When accessing the value the user selected through `select.values[0]`, you will only be able to retrieve the ID of the user, role, mentionable, or channel if you are not using a [`discord.ComponentType.string_select`](https://docs.pycord.dev/en/stable/api/enums.html#discord.ComponentType.string_select).
+
+:::
```
Comment (plun1331) [inline docs/interactions/ui-components/dropdowns.mdx:113]: This is incorrect, `Select.values` will resolve the received values to their respective objects, as they are sent with the interaction when selected.
https://github.com/Pycord-Development/pycord/blob/master/discord/ui/select.py#L328
https://docs.pycord.dev/en/master/api/ui_kit.html#discord.ui.Select.values

File: docs/interactions/ui-components/dropdowns.mdx [hunk 2]
```diff
@@ -100,6 +102,39 @@ In the `options` parameter, you pass a list of [`discord.SelectOption`](https://
...
+
+```python
+class MyView(discord.ui.View):
+    @discord.ui.select(
+        select_type=discord.ComponentType.user_select
+    )
+    async def select_callback(self, select, interaction):
+        await interaction.response.send_message(f"Hello, <@{select.values[0]}>")
```
Comment (plun1331) [inline docs/interactions/ui-components/dropdowns.mdx:121]:
```suggestion
        await interaction.response.send_message(f"Hello, {select.values[0].mention}")
```

File: docs/interactions/ui-components/dropdowns.mdx [hunk 3]
```diff
@@ -100,6 +102,39 @@ In the `options` parameter, you pass a list of [`discord.SelectOption`](https://
...
+When using a [`discord.ComponentType.channel_select`](https://docs.pycord.dev/en/stable/api/enums.html#discord.ComponentType.channel_select) type select menu, you can pass in a list of [`discord.ChannelType`](https://docs.pycord.dev/en/stable/api/enums.html#discord.ChannelType) values to limit which types of channels users can choose when interacting with the select menu.
+
+```python
+class MyView(discord.ui.View):
+    @discord.ui.select(
+        select_type=discord.ComponentType.channel_select,
+        channel_types=[discord.ChannelType.text, discord.ChannelType.voice]
+    )
+    async def select_callback(self, select, interaction):
+        await interaction.response.send_message(f"You selected <#{select.values[0]}>")
```
Comment (plun1331) [inline docs/interactions/ui-components/dropdowns.mdx:135]:
```suggestion
        await interaction.response.send_message(f"You selected {select.values[0].mention}")
```

**Review by plun1331 — CHANGES_REQUESTED** [review]
Consider also mentioning the shortcut decorators: https://docs.pycord.dev/en/master/api/ui_kit.html#discord.ui.string_select and below

File: docs/interactions/ui-components/dropdowns.mdx [hunk 4]
```diff
@@ -81,6 +81,8 @@ about making your code do amazing things, while Pycord handles the rest!
 
 #### Select Menu Properties
 
+-   `select_type`: The type of select to create. This must be a [`discord.ComponentType`](https://docs.pycord.dev/en/stable/api/enums.html#discord.ComponentType) value.
+-   `channel_types`: A list of channel types that can be selected in the menu. This is only valid for selects of `select_type` [`discord.ComponentType.channel_select`](https://docs.pycord.dev/en/stable/api/enums.html#discord.ComponentType).
 -   `options`\*: A list of [`discord.SelectOption`](https://docs.pycord.dev/en/stable/api/data_classes.html#discord.SelectOption) values. These are the options that can be selected in this menu.
```
Comment (plun1331) [inline docs/interactions/ui-components/dropdowns.mdx:86]: Can you make this not required as well kthx

**Review by plun1331 — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**